### PR TITLE
cli: make `jj workspace add` preserve all parents of current workspace

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -69,6 +69,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * The new template keywords `local_branches`/`remote_branches` are added to show
   only local/remote branches.
 
+* `jj workspace add` now preserves all parents of the old working-copy commit
```
Comment (ilyagr) [inline CHANGELOG.md:72]: Is this a new feature or a fixed bug?

**Comment by ilyagr** [comment]
I'm a little tired and really should read the tests, but what do we want the behavior of `jj workspace add` to be if the current working copy is non-empty? Should it duplicate the current commit? Create a child?

File: cli/src/commands/mod.rs [hunk 2]
```diff
@@ -2853,13 +2853,20 @@ fn cmd_workspace_add(
             .view()
             .get_wc_commit_id(old_workspace_command.workspace_id())
         {
-            tx.repo().store().get_commit(old_wc_commit_id)?.parents()[0].clone()
+            tx.repo().store().get_commit(old_wc_commit_id)?.parents()
```
Comment (yuja) [inline cli/src/commands/mod.rs:2856]: nit: `s/a parent/parents/` in the comment

**Review by yuja — APPROVED** [review]

File: cli/src/commands/mod.rs (hunk 2 above)
Comment (martinvonz) [reply cli/src/commands/mod.rs:2856]: Done.

File: CHANGELOG.md (hunk 1 above)
Comment (martinvonz) [reply CHANGELOG.md:72]: I consider the current behavior an oversight. Or perhaps it was poor judgment :)

**Comment by martinvonz** [comment]
> (Non-blocking) I'm a little tired and really should read the tests, but what do we want the behavior of `jj workspace add` to be if the current working copy is non-empty? Should it duplicate the current commit? Create a child?

It creates an empty commit that's a sibling to the current work-copy commit. That's unchanged by this PR. We can reconsider that, but I think the current behavior seems better than either of the other two behaviors you mention. If we duplicate the commit, then they'll most likely diverge and confuse the user. If we create it on top of the current working copy, then the new workspace will be more likely to require `jj workspace update-stale`.

File: CHANGELOG.md (hunk 1 above)
Comment (martinvonz) [reply CHANGELOG.md:72]: I'll mark this resolved, but let me know if you disagree. We can change it in this PR or later if necessary.

**Pull request merged**